man/systemd-measure.xml

```diff
@@ -61,11 +63,30 @@
       <varlistentry>
         <term><command>calculate</command></term>
 
-        <listitem><para>Pre-calculate the expected value seen in PCR register 11 after boot-up of a unified
+        <listitem><para>Pre-calculate the expected values seen in PCR register 11 after boot-up of a unified
         kernel image consisting of the components specified with <option>--linux=</option>,
         <option>--osrel=</option>, <option>--cmdline=</option>, <option>--initrd=</option>,
         <option>--splash=</option>, <option>--dtb=</option>, see below. Only <option>--linux=</option> is
-        mandatory.</para></listitem>
+        mandatory. (Alternatively, specify <option>--current</option> to use the current values of PCR
+        register 11 instead.)</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><command>sign</command></term>
+
+        <listitem><para>As with the <command>calculate</command> command, pre-calculate the expected value
+        seen in TPM2 PCR register 11 after boot-up of a unified kernel image. Then, cryptographically sign
+        the resulting values with the private/public key pair (RSA) configured via
+        <option>--private-key=</option> and <option>--public-key=</option>. This will write a JSON object to
+        standard output that contains signatures for all specified PCR banks (see
+        <option>--pcr-bank=</option>) below, which may be used to unlock encrypted credentials (see
+        <citerefentry><refentrytitle>systemd-creds</refentrytitle><manvolnum>1</manvolnum></citerefentry>) or
+        LUKS volumes (see
+        <citerefentry><refentrytitle>systemd-cryptsetup@.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>). This
+        allows binding secrets to a set of kernels for which such PCR 11 signatures can be provided.</para>
+
+        <para>Note that a TPM2 device must be available for this signing to take place, even though the
+        result is not tied to any TPM2 device or its state.</para></listitem>
```

Comment:
There's no technical reason to require a TPM here. It's mostly that I am not a TPM guru myself and what precisely the TPM hashes internally is a bit of a black box to me so far, so I just used the TPM as an implementation for the hashing, so that I don't have to figure out what precisely to hash. 

But your other comment should be enough to implement this without a TPM I guess. I'll put it on my TODO list.

Reply:
I guess the comment, to be accurate but perhaps confusing to non-tpm savvy users, is :that its for calculating the policy digest that is signed". I can whip you up a function that given a TPML_PCR_SELECTION will give you the hash if you'd like?

Reply:
oh, i'd be delighted for a patch/snippet doing that!

Reply:
@williamcroberts Any chance you might be willing to work on a patch to make systemd-measure work without a TPM device? Would be very useful for building images

Reply:
> @williamcroberts Any chance you might be willing to work on a patch to make systemd-measure work without a TPM device? Would be very useful for building images

Unfortunatley I got yanked off of all this work, so I won't have the cycles for major contributions, only support for previous work or guidance.

Reply:
@williamcroberts Actually I'm in luck, since https://github.com/systemd/systemd/pull/27627 was just opened, so the work has already been done!